Program/CVRPLIB.cpp

```diff
@@ -0,0 +1,78 @@
+//
+// Created by chkwon on 3/22/22.
+//
+
+#include <fstream>
+#include <cmath>
+#include "CVRPLIB.h"
+
+CVRPLIB::CVRPLIB(std::string pathToInstance) {
+
+	std::string content, content2, content3;
+	double serviceTimeData = 0.;
+
+	// Read INPUT dataset
+	std::ifstream inputFile(pathToInstance);
+	if (inputFile.is_open())
+	{
+		getline(inputFile, content);
+		getline(inputFile, content);
+		getline(inputFile, content);
+		for (inputFile >> content ; content != "NODE_COORD_SECTION" ; inputFile >> content)
+		{
+			if (content == "DIMENSION") { inputFile >> content2 >> nbClients; nbClients--; } // Need to substract the depot from the number of nodes
+			else if (content == "EDGE_WEIGHT_TYPE")	inputFile >> content2 >> content3;
+			else if (content == "CAPACITY")	inputFile >> content2 >> vehicleCapacity;
+			else if (content == "DISTANCE") { inputFile >> content2 >> durationLimit; isDurationConstraint = true; }
+			else if (content == "SERVICE_TIME")	inputFile >> content2 >> serviceTimeData;
+			else throw std::string("Unexpected data in input file: " + content);
+		}
+		if (nbClients <= 0) throw std::string("Number of nodes is undefined");
+		if (vehicleCapacity == 1.e30) throw std::string("Vehicle capacity is undefined");
+
+
+		x_coords = std::vector<double>(nbClients + 1);
+		y_coords = std::vector<double>(nbClients + 1);
+		demands = std::vector<double>(nbClients + 1);
+		service_time = std::vector<double>(nbClients + 1);
+
+		// Reading client coordinates
+		int custNum;
+		for (int i = 0; i <= nbClients; i++)
+			inputFile >> custNum >> x_coords[i] >> y_coords[i];
+
+		// Reading demand information
+		inputFile >> content;
+		if (content != "DEMAND_SECTION") throw std::string("Unexpected data in input file: " + content);
+		for (int i = 0; i <= nbClients; i++)
+		{
+			inputFile >> content >> demands[i];
+			service_time[i] = (i == 0) ? 0. : serviceTimeData ;
+		}
+
+		// Calculating 2D Euclidean Distance
+		// Rounding to integers will occur in Params.cpp
```

Comment:
The rounding convention is really a piece of information available in CVRPLib data file, I believe the calculation of the distances (including eventual rounding) should be made at this stage instead of later on in Params.cpp

Reply:
Make sense. A new commit follows.